integration/test/upgrade/upgrade_test.go

```diff
@@ -0,0 +1,57 @@
+// +build k8srequired
+
+package upgrade
+
+import (
+	"context"
+	"testing"
+
+	"github.com/giantswarm/apptest"
+
+	"github.com/giantswarm/app-exporter/integration/env"
+	"github.com/giantswarm/app-exporter/integration/key"
+	"github.com/giantswarm/app-exporter/integration/setup"
+	"github.com/giantswarm/app-exporter/pkg/project"
+)
+
+var (
+	config setup.Config
+)
+
+func init() {
+	var err error
+
+	{
+		config, err = setup.NewConfig()
+		if err != nil {
+			panic(err.Error())
+		}
+	}
+}
+
+func TestUpgrade(t *testing.T) {
+	var err error
+	ctx := context.Background()
+
+	currentApp := apptest.App{
+		CatalogName:   key.ControlPlaneCatalogName(),
+		Name:          project.Name(),
+		Namespace:     key.Namespace(),
+		WaitForDeploy: true,
+	}
+
+	desiredApp := apptest.App{
+		CatalogName:   key.ControlPlaneTestCatalogName(),
+		Name:          project.Name(),
+		Namespace:     key.Namespace(),
+		SHA:           env.CircleSHA(),
+		WaitForDeploy: true,
+	}
+
+	{
+		err = config.AppTest.UpgradeApp(ctx, currentApp, desiredApp)
```

Comment:
Only need to put the current App, desired App, and that's it.

Reply:
Nice, I like how clean the test wiring is.